tests/test_robotstxt_interface.py

```diff
@@ -141,6 +143,25 @@ def test_unicode_url_and_useragent(self):
         )
 
 
+class DecodeRobotsTxtTest(unittest.TestCase):
+    def test_native_string_conversion(self):
+        robotstxt_body = "User-agent: *\nDisallow: /\n".encode("utf-8")
+        decoded_content = decode_robotstxt(
+            robotstxt_body, spider=None, to_native_str_type=True
+        )
+        self.assertEqual(decoded_content, "User-agent: *\nDisallow: /\n")
+
+    def test_decode_utf8(self):
+        robotstxt_body = "User-agent: *\nDisallow: /\n".encode("utf-8")
+        decoded_content = decode_robotstxt(robotstxt_body, spider=None)
+        self.assertEqual(decoded_content, "User-agent: *\nDisallow: /\n")
+
+    def test_decode_non_utf8(self):
+        robotstxt_body = b"User-agent: *\n\xFFDisallow: /\n"
+        decoded_content = decode_robotstxt(robotstxt_body, spider=None)
+        self.assertNotEqual(decoded_content, "")
```

Comment:
Why not use `assertEqual` with the real returned value here?

Reply:
The idea behind this test case is that when a string is not in UTF-8 we can't reliably predict what the output will be as a result of adding `errors="ignore"` when decoding.
As it stands, we do obviously know what `robotstxt_body` will be and we might as well just use `assertEqual`, however I was trying to make sure that this code wouldn't fall under the [case where a `UnicodeDecodeError` is thrown](https://github.com/scrapy/scrapy/pull/6298/files#diff-0dc4cc282c7f78b4d6199a7b5de162f382efdf5d4836bb9fd567be94d15c8637R26-R36) - which just returns `""` - and I suppose we're trying to avoid that with this change?
Let me know what you think is best 🙂

Reply:
I also think we should use `assertEqual` with the actual value.

> was trying to make sure that this code wouldn't fall under the [case where a `UnicodeDecodeError` is thrown](https://github.com/scrapy/scrapy/pull/6298/files#diff-0dc4cc282c7f78b4d6199a7b5de162f382efdf5d4836bb9fd567be94d15c8637R26-R36)

`assertEqual` with the actual value works for this as well, as long as the actual value is not `""`, i.e. `self.assertEqual(decoded_content, "foo")` implies `self.assertNotEqual(decoded_content, "")`, but is better than the latter because it also proves we are using `ignore` and not some other error handling (e.g. replacing errors with special characters or something).

Reply:
Got it, will change it to `assertEqual` then 👍